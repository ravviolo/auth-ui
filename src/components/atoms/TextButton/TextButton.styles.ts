import styled from 'styled-components';

export const TextButton = styled.button`
  background: transparent;
  border: none;
  cursor: pointer;
`;
