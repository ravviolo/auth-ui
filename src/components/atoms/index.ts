export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
