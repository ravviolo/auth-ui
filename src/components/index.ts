export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
