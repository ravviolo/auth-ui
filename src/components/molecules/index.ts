export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
