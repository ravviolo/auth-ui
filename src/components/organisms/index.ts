export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
