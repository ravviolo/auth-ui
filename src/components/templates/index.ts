export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
