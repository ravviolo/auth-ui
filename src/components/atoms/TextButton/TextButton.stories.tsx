import { ComponentMeta, ComponentStory } from '@storybook/react';

import { TextButton } from 'components/atoms';

export default {
  title: 'Atoms/TextButton',
  component: TextButton,
  args: {
    label: 'Button',
    type: 'button',
  },
} as ComponentMeta<typeof TextButton>;

export const Default: ComponentStory<typeof TextButton> = (args) => (
  <TextButton {...args} />
);

Default.storyName = 'Text Button';
