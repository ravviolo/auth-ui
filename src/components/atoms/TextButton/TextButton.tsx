import * as S from './TextButton.styles';

export interface Props {
  label: string;
  type: 'button' | 'submit' | 'reset';
  testId: string;
  className?: string;
  onClick?: () => void;
}

export const TextButton = ({
  label,
  testId,
  className,
  type = 'button',
  onClick,
}: Props) => {
  return (
    <S.TextButton
      className={className}
      data-testid={testId}
      type={type}
      onClick={onClick}
    >
      {label}
    </S.TextButton>
  );
};
