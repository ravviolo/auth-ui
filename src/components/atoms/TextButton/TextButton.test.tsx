import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';

import { TextButton } from 'components/atoms';

import { Props as TextButtonProps } from './TextButton';

describe('TextButton', () => {
  const mockOnClick = jest.fn();

  const props: TextButtonProps = {
    label: 'Test Button',
    testId: 'test-text-btn',
    type: 'button',
    onClick: mockOnClick,
  };

  it('renders button with text received through props', () => {
    render(<TextButton {...props} />);

    const textBtn = screen.getByTestId('test-text-btn');

    expect(textBtn).toHaveTextContent('Test Button');
  });

  it('handles click event, fires event handler once', async () => {
    render(<TextButton {...props} />);

    const textBtn = screen.getByTestId('test-text-btn');

    await userEvent.click(textBtn);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const tree = create(<TextButton {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
