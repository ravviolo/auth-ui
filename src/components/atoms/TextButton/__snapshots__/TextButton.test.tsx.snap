// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton matches snapshot 1`] = `
<button
  className="sc-jqUVSM ebpvV"
  data-testid="test-text-btn"
  onClick={[MockFunction]}
  type="button"
>
  Test Button
</button>
`;
